amptorch/trainer.py

```diff
@@ -76,10 +77,20 @@ def load_dataset(self):
         )
 
         # TODO: Allow for alternative fingerprinting schemes
-        fp_params = self.config["dataset"]["fp_params"]
-        self.descriptor = Gaussian(Gs=fp_params, elements=self.elements)
-        self.forcetraining = self.config["model"].get("get_forces", True)
+        IMPLEMENTED_FP_SCHEMES = ["gaussian", "mcsh"]
+        fp_scheme = self.config["dataset"].get("fp_scheme", "gaussian").lower()
+        
+        if fp_scheme == "gaussian":
+            fp_params = self.config["dataset"]["fp_params"]
+            self.descriptor = Gaussian(Gs=fp_params, elements=self.elements)
+        elif fp_scheme == "mcsh":
+            fp_params = self.config["dataset"]["fp_params"]
+            self.descriptor = AtomisticMCSH(MCSHs=fp_params, elements=self.elements)
+        else:
+            raise NotImplementedError
```

Comment:
Let's move this logic to the `AtomsDataset` instead. The `descriptor` argument for `AtomsDataset` can now be a string  in which this above logic is applied. We want to keep the trainer as clean as possible.

Reply:
OK, I'll make the change

Reply:
Actually, I think we need to clarify this a bit, 
if we want to move this logic to be in AtomsDataset, we need to add descriptor_type, descriptor_config and elements as extra arguments to that object, and somehow I think this is not as clean as just taking an descriptor object
also the trainer won't have access to the descriptor (tho we might not need it to anyway)

Reply:
Or, actually, maybe we can put this part into util, which I think it's the cleanest way

Reply:
Maybe just make a method in dataset.py instead, like I did with the collater - so we centralize methods/classes rather than putting it in utils.